### PR TITLE
Updating course_structure

### PR DESCRIPTION
**Pull request by jordan-springer**
# Description

Please review `dim_course_structure` updates. These updates are a re-factor of some [development code](https://github.com/code-dot-org/analytics-infrastructure/pull/83) @nataliazm99 is looking to promote to production in `analysis`. Expectation is a 1-1 match between these two models for consistency around defining course structures.

## Links

Jira ticket(s): [dataops-559](https://codedotorg.atlassian.net/browse/DATAOPS-559)

## Testing story

- [x] Does your change include appropriate tests on key columns?
      eg.
      - `not_null`
      - `unique`
      - `dbt_utils.unique_combination_of_columns: , ["value","value","value"...]

**Note: when submitting a new model for review please make sure the following have been tested:**

1. The model compiles (`dbt build -m 'your_model'`)
         or: _has the dbt Cloud job succeeded?_
3. The model runs (`dbt run -m 'your_model'`)
4. The model produces accessible data in the DW (`select 1 from 'your_model'`)

## Privacy

- [ ] 1.	Does this change involve the collection, use, or sharing of new Personal Data?
- [x] 2.    Do these data exist in the appropriate schema(s)? 
- [ ] 3.	Does this change involve a new or changed use or sharing of existing Personal Data?
- [ ] 4.    Consider: will this data be visible on Tableau? will this data be surfaced in a report exported from Trevor?
- [ ] 5.    If yes to any of the above, please list the models, columns, and justification below:
      i.
      ii.
      iii. 

## PR Checklist:
**Note: if these are not all checked, the PR will be sent back.**

- [x] Tests provide adequate coverage
- [x] Privacy and Security impacts have been assessed
- [x] Code adheres to [style guide](https://docs.getdbt.com/best-practices/how-we-style/0-how-we-style-our-dbt-projects)👀 and is **[DRY](https://docs.getdbt.com/terms/dry)**
- [x] Code is well-commented (**please do not leave extraneous commentary in model code, if it is for the purpose of documentation, please relocate accordingly)
- [x] Appropriate documentation has been provided (see `.yml.`, did `dbt docs generate` succeed?)
- [x] New features are translatable or updates will not break up/downstream models
- [x] Relevant documentation has been added or updated (i.e. `dbt docs` has been updated successfully on [Github Pages](code-dot-org.github.io/analytics/)
- [x] Pull Request is labeled appropriately (eg. `chore/`, `feature/`, `fix/`)
- [x] Follow-up work items (including potential tech debt) are tracked and linked (if applicable)
* make sure Rosetta() procedure matches updates (see: https://codedotorg.atlassian.net/browse/DATAOPS-558)

### CODE REVIEW

**Review by jordan-springer — COMMENTED** [review]
ready for your review @nataliazm99

**Review by nataliazm99 — CHANGES_REQUESTED** [review]
I left comments in code, and here are some other findings:
1) Field `lesson_unplugged` is missing

2) Could be useful to flag levels who have sublevels by left joining `dashboard_production.parent_levels_child_levels` and adding these fields:
 ```
     , case when plcl.parent_level_id is not null then 'Y' else 'N' end parent_level
      , plcl.kind parent_level_kind
```

3) Differences with `dashboard.public.nzm_view_course_structure_redesigned`:
1. Assessments: has nulls for 2,865 records, should be 0
2. Course_name_long and course_name_short has nulls for 2,173 records that have values in my view
3. level_script_order is different for 594 records, e.g. level_id in (45530, 45229) 78 and 39 in my view, 77 and 38 in this model.

File: dbt/models/marts/courses/dim_course_structure.sql [hunk 1]
```diff
@@ -49,8 +59,8 @@ script_names as (
 combined as (
     select 
         -- courses
-        ug.unit_group_id as course_id,
-        ug.name as course_name,
+        ug.unit_group_id    as course_id,
+        ug.unit_group_name  as course_name,
         cn.course_name_short,
         cn.course_name_long,
         sc.course_name_true,
```
Comment (jordan-springer) [inline dbt/models/marts/courses/dim_course_structure.sql:66]: outdated logic, will remove

**Comment by jordan-springer** [comment]
Per review with NZM:
- remove `course_name_true` logic (see `case` statement)
- ensure we are not bringing in any _child_ levels (see last `left join`)

**Comment by nataliazm99** [comment]
- remove course_name_true logic for individual scripts. Use this case statement:

```
,
            CASE
	            WHEN json_extract_path_text(sc.properties, 'curriculum_umbrella', true) = '' THEN 'other'
	            ELSE LOWER(json_extract_path_text(sc.properties, 'curriculum_umbrella', true)) 
            END AS course_name_true,
```

- join `contained_levels` but only to indicate if it is a group level, don't bring in any child levels:
     1. add join statement: 
```            LEFT JOIN dashboard_production.contained_levels cl on lsl.level_id = cl.level_group_level_id 
```
2. add to the select:
```
             case
            	when cl.level_group_level_id is not null then 'Y'
            	else 'N' 
            end is_group_level,
            cl.contained_level_type group_level_type
```

**Review by nataliazm99 — CHANGES_REQUESTED** [review]
Overall it's good, matches the logic in the view and no meaningful changes in downstream models `dim_user_course_activity` and `dim_school_status`
The only issue that needs addressed is that field `lesson_unplugged` is still missing

**Pull request merged**